File: treebolicOwlServices/src/main/java/org/treebolic/owl/service/TreebolicOwlBroadcastService.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.treebolic.owl.service;

import android.content.Context;

import org.treebolic.services.IModelFactory;
import org.treebolic.services.TreebolicBroadcastService;

import java.io.IOException;

import androidx.annotation.NonNull;

/**
 * Treebolic Owl broadcast service
 */
public class TreebolicOwlBroadcastService extends TreebolicBroadcastService
{
	/**
	 * Constructor
	 */
	public TreebolicOwlBroadcastService()
	{
		super();
	}

	@Override
	protected IModelFactory createModelFactory(@NonNull final Context context) throws IOException
	{
		return new ModelFactory(context);
	}

	@SuppressWarnings("SameReturnValue")
	@NonNull
	@Override
	public String getUrlScheme()
	{
		return "owl:";
	}
}

File: treebolicOwlServices/src/main/java/org/treebolic/owl/service/TreebolicOwlBroadcastService.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.owl.service

import android.content.Context
import org.treebolic.services.IModelFactory
import org.treebolic.services.TreebolicBroadcastService
import java.io.IOException

/**
 * Treebolic Owl broadcast service
 */
class TreebolicOwlBroadcastService : TreebolicBroadcastService() {

    @Throws(IOException::class)
    override fun createModelFactory(context: Context): IModelFactory {
        return ModelFactory(context)
    }

    override val urlScheme: String
        get() = "owl:"
}
